File: two_phase_flow_complex_domain/post_process/da1.m
function [r,z,u,v,p,f,f2, t]=da1(file,m,n)
    
    ff_info = dir(file);
    ff_info.date

    fid=fopen(file,'r');
    fseek(fid,4,'cof');
    r=fread(fid,[n,m],'double');
    fseek(fid,8,'cof');
    z=fread(fid,[n,m],'double');
    fseek(fid,8,'cof');
    u=fread(fid,[n,m],'double');
    fseek(fid,8,'cof');
    v=fread(fid,[n,m],'double');
    fseek(fid,8,'cof');
    p=fread(fid,[n,m],'double');
    fseek(fid,8,'cof');
    f=fread(fid,[n,m],'double');

    fseek(fid,8,'cof');
    f2=fread(fid,[n,m],'double');

    fseek(fid,8,'cof');
    t=fread(fid,1,'double');
    fclose(fid);

end

File: two_phase_flow_complex_domain/post_process/da1.py
import numpy as np
import os
import datetime
import struct


def da1(file, m, n):
    # 显示文件最后修改时间（类似 MATLAB 中的 ff_info.date）
    mod_time = os.path.getmtime(file)
    print("Last modified:", datetime.datetime.fromtimestamp(mod_time))

    with open(file, 'rb') as fid:
        fid.seek(4, 1)  # Skip 4 bytes
        r = np.fromfile(fid, dtype=np.float64, count=m * n).reshape((n, m))

        fid.seek(8, 1)
        z = np.fromfile(fid, dtype=np.float64, count=m * n).reshape((n, m))

        fid.seek(8, 1)
        u = np.fromfile(fid, dtype=np.float64, count=m * n).reshape((n, m))

        fid.seek(8, 1)
        v = np.fromfile(fid, dtype=np.float64, count=m * n).reshape((n, m))

        fid.seek(8, 1)
        p = np.fromfile(fid, dtype=np.float64, count=m * n).reshape((n, m))

        fid.seek(8, 1)
        f = np.fromfile(fid, dtype=np.float64, count=m * n).reshape((n, m))

        fid.seek(8, 1)
        f2 = np.fromfile(fid, dtype=np.float64, count=m * n).reshape((n, m))

        fid.seek(8, 1)
        t = struct.unpack('d', fid.read(8))[0]  # Read a single double

    return r, z, u, v, p, f, f2, t